src/server/middleware/errorHandler.js

```diff
@@ -2,6 +2,9 @@ const { error } = require("winston");
 const PropsGenerator = require("../propsGenerator");
 
 const errorHandler = (err, req, res, next) => {
+  if (res.headersSent) {
+    return next(err);
+  }
   error(err);
```

Comment:
I think it's OK to call `error(err);` before we return, just so the main error is logged. We have to call `return next(err);` before any of the lines accessing res since that is what triggers the secondaary error